src/components/badge/badge.style.ts

```diff
@@ -14,6 +14,30 @@ const StyledCounter = styled.div`
   margin-top: -1px;
 `;
 
+const StyledCounterSpan = styled.span`
```

Comment:
As @mkrds points out there's a lot of shared styling here, I think we could just use the `as` prop here to achieve polymorphism, when there's an onClick render a `Button` and when there's not a `span`. This will allow you to conditionally apply any diverged styles whilst not repeating the shared parts etc

Reply:
You could do this in the component or use `attrs` to achieve it in the styled component itself

```
// renamed StyledButton to be more generic
const StyledBadge = styled.span.attrs(({ onClick }) => ({
  as: onClick ? Button : undefined,
}))`
```